app/controllers/chat_channel_memberships_controller.rb

```diff
@@ -49,12 +49,19 @@ def create
     redirect_to "/chat_channel_memberships/#{@chat_channel.chat_channel_memberships.where(user_id: current_user).first&.id}/edit"
   end
 
-  def remove_invitation
+  def remove_membership
     @chat_channel = ChatChannel.find(params[:chat_channel_id])
     authorize @chat_channel, :update?
-    flash[:settings_notice] = "Invitation Removed."
-    ChatChannelMembership.where(chat_channel_id: @chat_channel.id, id: params[:invitation_id], status: "pending").first&.destroy
-    redirect_to "/chat_channel_memberships/#{@chat_channel.chat_channel_memberships.where(user_id: current_user).first&.id}/edit"
+    @chat_channel_membership = ChatChannelMembership.find(params[:membership_id])
+    if params[:status] == "pending"
+      @chat_channel_membership.destroy
+      flash[:settings_notice] = "Invitation Removed."
+    else
+      @chat_channel_membership.update(status: "removed_from_channel")
+      @chat_channel_membership.remove_from_index!
+      flash[:settings_notice] = "Removed #{@chat_channel_membership.user.name}"
+    end
+    redirect_to "/chat_channel_memberships/#{ChatChannelMembership.where(chat_channel_id: params[:chat_channel_id], user_id: current_user).first&.id}/edit"
```

Comment:
I know you're just moving code around here, but I think it'd be good if we start using Rails' path helpers more instead of manually constructing URLs.

```ruby
chat_channel_membership = ChatChannelMembership.where(....)
redirect_to edit_chat_channel_membership_path(chat_channel_membership)
```

Reply:
updated all manually constructed url with rails path helpers in chat_channel_memberships_controller.rb